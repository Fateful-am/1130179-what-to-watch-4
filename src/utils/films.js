export const MovieInfoMode = {
  CARD: `card`,
  PROMO: `promo`,
  OVERVIEW: `overview`,
  DETAIL: `detail`,
  REVIEWS: `reviews`
};

export const getMovies = (sourceMovies, sliceFrom, count) => {
  return sourceMovies.slice(sliceFrom, sliceFrom + count).map((movie) => {
    return {
      id: movie.id,
      title: movie.title,
      posterImage: movie.posterImage,
    };
  });
};

export const getMoviePromoMode = (sourceMovies, index) => {
  const movie = sourceMovies[index];
  return {
    id: movie.id,
    title: movie.title,
    genre: movie.genre,
    posterImage: movie.posterImage,
    backgroundImage: movie.backgroundImage,
    released: movie.released
  };
};
