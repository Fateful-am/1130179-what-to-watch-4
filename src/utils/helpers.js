import moment from 'moment';

export const convertToImageName = (movieName) => {
  return movieName.toLowerCase().split(` `).join(`-`).split(`:`).join(``);
};

export const getMovieById = (movies, id) => {
  const movieIndex = movies.findIndex((movie) => movie.id === id);
  if (movieIndex > -1) {
    return movies[movieIndex];
  }
  return null;
};

export const formatDuration = (duration) => {
  if (duration < 60) {
    return moment.utc(moment.duration(duration, `minutes`).asMilliseconds()).format(`m[m]`);
  }
  return moment.utc(moment.duration(duration, `minutes`).asMilliseconds()).format(`H[h] m[m]`);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};


export const smallMovieCards = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`
];
