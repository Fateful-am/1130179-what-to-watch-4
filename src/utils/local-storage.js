import {extend} from './helpers';

const ItemsType = {
  AUTH_STATUS: `authStatus`,
  LAST_URL: `lastUrl`,
};

export default class LocalStorage {
  constructor(key, storage) {
    this._storage = storage;
    this._storeKey = key;
  }

  _getItems() {
    try {
      return JSON.parse(this._storage.getItem(this._storeKey)) || {};
    } catch (err) {
      return {};
    }
  }

  _getItem(item) {
    const store = this._getItems();
    return store[item];
  }

  _setItem(item, value) {
    const store = this._getItems();
    this._storage.setItem(
        this._storeKey,
        JSON.stringify(extend(store, {
          [item]: value,
        }))
    );
  }

  getAuthStatus() {
    return this._getItem(ItemsType.AUTH_STATUS);
  }

  setAuthStatus(authStatus) {
    this._setItem(ItemsType.AUTH_STATUS, authStatus);
  }

  getLastUrl() {
    return this._getItem(ItemsType.LAST_URL) || `/`;
  }

  setLastUrl(url) {
    this._setItem(ItemsType.LAST_URL, url);
  }
}
