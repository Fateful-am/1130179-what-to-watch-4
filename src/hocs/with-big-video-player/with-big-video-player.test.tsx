import * as React from 'react';
import * as renderer from 'react-test-renderer';
import withBigVideoPlayer from './with-big-video-player';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import NameSpace from '../../reducer/name-space';
import {TEST_DATA} from '../../utils/test-data';
import {Provider} from 'react-redux';
import {noop} from '../../utils/helpers';

interface Props {
  children: React.ReactNode | React.ReactNode[];
}

Enzyme.configure({
  adapter: new Adapter(),
});

const mockStore = configureStore([]);

const MockComponent: React.FunctionComponent<Props> = (props: Props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const store = mockStore({
  [NameSpace.DATA]: TEST_DATA.initialStoreDataState,
});

const MockComponentWrapped = withBigVideoPlayer(MockComponent);

it(`withBigVideoPlayer is rendered correctly`, () => {
  const tree = renderer.create((
    <Provider store={store}>
      <MockComponentWrapped
        videoLink={``}
        previewImage={``}
        title={``}
        onExitButtonClick={noop}
        match={{params: {id: `8`}}}
      />
    </Provider>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});

