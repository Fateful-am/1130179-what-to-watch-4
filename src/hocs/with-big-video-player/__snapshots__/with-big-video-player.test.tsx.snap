// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withBigVideoPlayer is rendered correctly 1`] = `
<div>
  <video
    className="player__video"
  />
</div>
`;
